### ISSUE TITLE
Automated Social Posting (500,000 Sat Bounty)

### ISSUE THREAD

**Issue by ghost** [stackernews/stacker.news#434]
**Is your feature request related to a problem? Please describe.**
Stacker News manually posts top stories and regular platform stats to Twitter and Nostr, but many of these posts follow a similar framework and automation could save a lot of time and improve the reaction time of these posts.

**Describe the solution you'd like**
The completion of this task involves creating a script that will monitor Stacker News and share links to top stories and platform metrics following the cadence and rules outlined below.

There are 4 types of posts that should be automated for this task:
- New top stories
- Large upcoming rewards
- Top stories of the week
- Top stackers of the week

New top stories
- Every time a new story hits the top of Stacker News, it should be shared on Twitter and Nostr. This should only occur the first time a new story hits the top spot on the site. Stacker News Ads should be ignored if/when they reach the top spot. If a post in the top spot falls to the second spot and re-gains the first spot later on, it should not be posted on Twitter and Nostr a second time. The social post should use the following framework:

🔥 Trending on SN 🔥

[Post Title]
[Post Link with @sn referral tag]

Large upcoming rewards
- At 4pm CT every day, a social post on Nostr and Twitter should be made if the rewards pool is larger than 125k sats. The social post should use the following framework:

In 8 hours, we're giving away over [insert number] sats to all the best zappers on Stacker News!

[Rewards Page Link with @sn referral tag]

Top stories of the week
- Every Saturday at 12pm CT, a social post on Nostr and Twitter should be made to draw attention to the top stories of the week. The social post should use the following framework:

This week [Number of stackers who created a post this week] stackers created [number of posts this week] Stacker News posts, catch up on all the top stories from the last week!

[Top Posts of the Week By Votes Link with @sn referral tag]

Top stackers of the week
- Every Sunday at 12pm CT, a social post on Nostr and Twitter should be made to draw attention to the top stackers of the week. The social post should use the following framework:

This week [Number of stackers who stacked sats] stackers earned [number of sats stacked this week] sats, who topped this week's leaderboard?

🥇 [top stacker nym]: [sats stacked by top stacker this week]
🥈 [second stacker nym]: [sats stacked by second stacker this week]
🥉 [third stacker nym]: [sats stacked by third stacker this week]

Full leaderboard 👇

[Top Stackers of the Week By Sats Stacked Link with @sn referral tag]

**Comment by rleed** [comment]
Where do you currently get # stackers created # posts, and # stackers earned # sats?

**Comment by kerooke** [comment]
@rleed Those numbers aren't explicitly available on the site right now... but could be a good addition to the Analytics page.

The last 3 numbers can be determined by looking at the Analytics page: https://stacker.news/stackers/week, and the first number could be determined by looking at the unique accounts on the RSS feed.

**Comment by ekzyis** [comment]
> Where do you currently get # stackers created # posts, and # stackers earned # sats?
> 
> @rleed Those numbers aren't explicitly available on the site right now... but could be a good addition to the Analytics page.
> 
> The last 3 numbers can be determined by looking at the Analytics page: https://stacker.news/stackers/week, and the first number could be determined by looking at the unique accounts on the RSS feed.

Don't forget that you are not limited to what the frontend makes available. You can always directly talk to the database.

**Comment by rleed** [comment]
> > > Where do you currently get # stackers created # posts, and # stackers earned # sats?
> > 
> > 
> > @rleed Those numbers aren't explicitly available on the site right now... but could be a good addition to the Analytics page.
> > The last 3 numbers can be determined by looking at the Analytics page: https://stacker.news/stackers/week, and the first number could be determined by looking at the unique accounts on the RSS feed.
> 
> Don't forget that you are not limited to what the frontend makes available. You can always directly talk to the database.

Yeah, I found everything in the GQL API.

**Comment by rleed** [comment]
> The completion of this task involves creating a script that will monitor Stacker News and share links to top stories and platform metrics following the cadence and rules outlined below.

FWIW, the complete bash solution I submitted does seem in line with the original wording.

**Comment by ekzyis** [comment]
> FWIW, the complete bash solution I submitted does seem in line with the original wording.

You're right, not sure if that was intended. Let's wait for @huumn's opinion on this
